src/services/side-panels/side-panel-item.ts

```diff
@@ -0,0 +1,47 @@
+import {FunctionalComponent, RefObject} from 'preact';
+import {Toggle} from './components/side-panel.component';
+import {ui} from 'kaltura-player-js';
+const {ReservedPresetNames} = ui;
+
+export type SidePanelPosition = 'top' | 'bottom' | 'right' | 'left';
+export type SidePanelMode = 'alongside' | 'hidden' | 'over';
+export type ReservedPresetName = 'Playback' | 'Live';
+
+export class SidePanelItem {
+  public renderIcon?: FunctionalComponent;
+  public renderContent: FunctionalComponent;
+  public presets: ReservedPresetName[];
+  public position: SidePanelPosition;
+  public expandMode: SidePanelMode;
+  public hooks: {onActivate: () => void; onDeactivate: () => void};
+
+  constructor(item: SidePanelItem) {
+    this.renderIcon = item.renderIcon;
+    this.renderContent = item.renderContent;
+    this.presets = item.presets || Object.values(ReservedPresetNames);
+    this.position = item.position;
+    this.expandMode = item.expandMode;
+    this.hooks = {
+      onActivate: (): void => {
+        return;
```

Comment:
() => {} would be shorter

Reply:
Unfortunately i have no choice due to the linter
@typescript-eslint/no-empty-function